toolbelt/generate-configs.js

```diff
@@ -3,29 +3,43 @@
  */
 
 const fs = require( 'fs' );
-const packageJsonTemplate = require( './templates/package.json' );
+const path = require( 'path' );
 
 const packageList = require('./packages-list.js').packageList();
 
 console.log('Generating package.json and tsconfig.json for:');
 
+const packagesRootDir = path.resolve(__dirname, '..', 'packages');
+const templatesRootDir = path.resolve(__dirname, 'templates');
+
+const packageJsonTemplate = require( path.resolve(templatesRootDir, 'package.json') );
+const tsconfigJsonTemplate = require( path.resolve(templatesRootDir, 'tsconfig.json') );
+
 for (let packageName of packageList) {
 
   console.log(packageName);
 
-  fs.createReadStream('./toolbelt/templates/tsconfig.json').pipe(fs.createWriteStream(`./packages/${packageName}/tsconfig.json`));
+  const packageRootDir = path.resolve(packagesRootDir, packageName);
+
+  // Copy tsconfig.json template to each package folder
+  fs.writeFileSync(path.resolve(packageRootDir, 'tsconfig.json'), JSON.stringify(tsconfigJsonTemplate, null, 2) + "\n")
 
-  const originalPackageJson = require(`../packages/${packageName}/package.json`);
+  // Generate package.json file base on package package.json file and template
+  const packagePackageJsonPath = path.resolve(packageRootDir, 'package.json');
+
+  const originalPackageJson = (
+    fs.existsSync( packagePackageJsonPath ) ? require( packagePackageJsonPath ) : {}
+  );
 
   const packageJson = Object.assign({}, packageJsonTemplate, {
-    name: `@blaze-elements/${packageName}`,
-    version: originalPackageJson.version || '1.0.0'
+    name: originalPackageJson.name || `@blaze-elements/${packageName}`,
+    version: originalPackageJson.version || '1.0.0',
+    dependencies: originalPackageJson.dependencies || undefined
   });
 
   const packageJsonString = (JSON.stringify(packageJson, null, 2) + "\n").replace(/packageName/g, packageName);
```

Comment:
Saving of the package JSON could be extracted to a separate function.

Also, why do we need **"\n"** at the end - is it really necessary?